### ISSUE TITLE
NUnit4 and legacy MSBUILD

### ISSUE THREAD

**Issue by Gakk** [nunit/nunit#4579]
I have a .NET Framework project that has to build with legacy `msbuild`, and can not use `dotnet build` or `dotnet msbuild` because they does not support the task `ResolveCOMReference`. Ref: [MSBuild MSB4803 Error](https://learn.microsoft.com/nb-no/visualstudio/msbuild/resolvecomreference-task?view=vs-2022#msb4803-error).

When using NUnit 3 together with `nunit3-console` everything works fine, but when upgrading to NUnit 4 it does not work, and results in the following error:

```
1) Invalid : .\bin\Debug\net48\NUnit4Test.dll
No suitable tests found in '.\bin\Debug\net48\NUnit4Test.dll'.
Either assembly contains no tests or proper test driver has not been found.
```

I am unsure where the issue might be:

 - Is NUnit4 not working with legacy `msbuild`?
 - Is NUnit4 not compatible with `NUnit3TestAdapter`?
 - Is NUnit4 not supported by `nunit3-console`?

I have created a minimal example with a detailed README and GitHub Workflow that demonstrates the issue:
https://github.com/Gakk/Nunit4Test

And to follow-up; will there be released a `NUnit4TestAdapter` or `nunit4-console`?

**Comment by OsirisTerje** [comment]
Hi!  
The answer to all your questions are YES, it works.

Now:  The adapter must be at least version 4.5.0.  You have that.

But, the console must be version 3.15.5.   The 3.15.X series is the one that is compatible with NUnit 4, and the one that is "going forward".  

We are struggling with the 3.16.X series, so will probably port reliable fixes from that one over to the 3.15.X series (and probably make that into a 3.17 version).

Unfortunately we don't have any dedicated people on the console/engine now, so it is slow moving.

About the naming, that is an unfortunate earlier name that was done here.  We will not do more of those.  Going forward we will follow Semver, and that means you will see major NUnit version much more often, every time there is some kind of breaking change. We need to keep updateable, so we are stuck with the naming we have now. 

NUnit version 4 is also the same as NUnit version 3 (as compared to NUnit version2), but with breaking changes and a years long list of other changes.

**Comment by OsirisTerje** [comment]
Yes, will do that :-)  Thanks !
